### PR TITLE
Drop method `status_code_error`

### PR DESCRIPTION
**Pull request by Ichimonji10**
The method is quite complicated and only used in one location. Test results
against a downstream RHEL 7.1 system:

    $ nosetests tests/foreman/api/test_activationkey.py -m test_get_releases_status_code
    .
    ----------------------------------------------------------------------
    Ran 1 test in 7.943s

    OK

### CODE REVIEW

**Comment by Ichimonji10** [comment]
Pylint is happy with the `tests/foreman/api/test_activationkey.py`. I ran across this issue while performing a separate refactoring and decided to submit it as a separate PR.

File: tests/foreman/api/test_activationkey.py [hunk 1]
```diff
@@ -222,11 +221,7 @@ def test_get_releases_status_code(self):
             verify=False,
         )
         status_code = httplib.OK
-        self.assertEqual(
-            status_code,
-            response.status_code,
-            status_code_error(path, status_code, response),
-        )
+        self.assertEqual(status_code, response.status_code)
```
Comment (JacobCallahan) [inline tests/foreman/api/test_activationkey.py]: :+1:

**Comment by JacobCallahan** [comment]
"1 addition and 93 deletions"! ACK

**Comment by oshtaier** [comment]
ACK

**Comment by elyezer** [comment]
:+1:

**Pull request merged**